Dockerfile

```diff
@@ -7,8 +7,25 @@ RUN npm install -g langfuse-langchain@3.18.0
 RUN npm install -g typescript@5.6.2
 RUN npm install -g grunt
 
-# Install bash in the existing base image
-RUN apk update && apk add --no-cache bash
-RUN chown -R node:node /usr/local/lib /usr/local/bin
+# Install bash and OpenSSL for certificate generation
+RUN apk update && apk add --no-cache bash openssl
+
+# Create directory for certificates
+RUN mkdir -p /home/node/certificates
+WORKDIR /home/node/certificates
+
+# Generate self-signed certificate
+RUN openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
+    -keyout n8n-key.pem -out n8n-cert.pem \
+    -subj "/C=US/ST=State/L=City/O=Organization/CN=localhost"
+
+# Set proper ownership for directories and files
+RUN chown -R node:node /usr/local/lib /usr/local/bin /home/node/certificates
```

Comment:
You could use `USER node` on line 13 to avoid changing the ownership of the dir and files created during the openssl execution

Reply:
Done